x-pack/plugins/security_solution/cypress/e2e/detection_rules/maintenance_window.cy.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { INTERNAL_BASE_ALERTING_API_PATH } from '@kbn/alerting-plugin/common';
+import { cleanKibana } from '../../tasks/common';
+import { login, visit } from '../../tasks/login';
+import { DETECTIONS_RULE_MANAGEMENT_URL } from '../../urls/navigation';
+
+describe('Maintenance window callout on Rule Management page', () => {
+  let maintenanceWindowId = '';
+
+  before(() => {
```

Comment:
It should work with async functions so Cypress will wait for the maintenance window to be created otherwise it can be flaky. I haven't tried async callbacks in `before` but it worth to check it out.